includes/models/class-payment.php

```diff
@@ -77,6 +77,8 @@ protected function before_save( $new = false ) {
                 $this->status = 'completed';
             }
         }
+        
+        $this->maybe_set_test_mode();
```

Comment:
I am seeing an issue here. It looks like this is running every time a payment is saved. So when I go to an existing payment and click "save payment", the payment mode gets changed. 

Is there a spot we could add this so it only gets added when the payment is created?

Reply:
@stephywells I can probably use the $new param that is passed to the before_save method if that presents a new payment. I will check it what exactly is sent there.